### PR TITLE
Fix parsing embedded bitmap resources

### PR DESCRIPTION
**Pull request by TerryFogg**
## Description
This adds the "GetBitmap" from a resource file.
This code has been available but due to incorrect identification of resource kind was not working.

## Motivation and Context
Fixes a problem with bitmap resources.

## How Has This Been Tested?
From an nf C# program add a nf Resource file.
Add Bitmaps of any type to the resource file.
Check that an entry in the resource file designer code has a new GetBitmap(...) and resource id in the file.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [x] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [ ] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:
- [x] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/nanoframework/nf-Visual-Studio-extension?pullRequest=729) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/nanoframework/nf-Visual-Studio-extension?pullRequest=729) before we can accept your contribution.<br/><br/>:x: TerryFogg [sign now](https://cla.dotnetfoundation.org/nanoframework/nf-Visual-Studio-extension?pullRequest=729)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/nanoframework/nf-Visual-Studio-extension?pullRequest=729) it.</sub>

File: Tools.BuildTasks-2019/ProcessResourceFiles.cs [hunk 1]
```diff
@@ -1314,91 +1326,46 @@ private void Compress1bpp(NanoResourceFile.CLR_GFX_BitmapDescription bitmapDescr
...
-                    case PixelFormat.Format48bppRgb:
-                    case PixelFormat.Format16bppRgb555:
-                    case PixelFormat.Format16bppRgb565:
-                        bitsPerPixel = 16;
-                        formatDst = PixelFormat.Format16bppRgb565;
-                        break;
-                    default:
-                        throw new NotSupportedException($"PixelFormat ({bitmap.PixelFormat.ToString()}) of '{Name}' resource not supported. Only 1bpp and 16bpp.");
-                }
+            private byte[] GetBitmapDataBmp(Bitmap bitmap, out NanoResourceFile.CLR_GFX_BitmapDescription bitmapDescription)
```
Comment (josesimoes) [inline Tools.BuildTasks-2019/ProcessResourceFiles.cs:1339]: Noted about this. Please remove the comment.

File: Tools.BuildTasks-2019/ProcessResourceFiles.cs [hunk 2]
```diff
@@ -891,65 +891,77 @@ public static ResourceTypeDescription ResourceTypeDescriptionFromResourceType(by
...
+                        {
+                            bitmapImage = Image.FromStream(new MemoryStream(rawValue)) as Bitmap;
+                            imageJPeg = bitmapImage.RawFormat.Equals(ImageFormat.Jpeg);
+                            imageGif = bitmapImage.RawFormat.Equals(ImageFormat.Gif);
+                            imageBmp = bitmapImage.RawFormat.Equals(ImageFormat.Bmp);
+                            imageIcon = bitmapImage.RawFormat.Equals(ImageFormat.Icon);
+                        }
+                        catch
+                        {
+                        } // Ignore error if not a bitmap
```
Comment (josesimoes) [inline Tools.BuildTasks-2019/ProcessResourceFiles.cs:920]: comment inside catch block please

File: Tools.BuildTasks-2019/ProcessResourceFiles.cs [hunk 3]
```diff
@@ -891,65 +891,77 @@ public static ResourceTypeDescription ResourceTypeDescriptionFromResourceType(by
 
             public static Entry CreateEntry(string name, object value, string defaultNamespace, string defaultDeclaringClass)
             {
-                string stringValue = value as string;
-                Bitmap bitmapValue = value as Bitmap;
-                Icon iconValue = value as Icon;
-                byte[] rawValue = value as byte[];
 
                 Entry entry = null;
+                string nameOfResourceType = value.GetType().Name;
```
Comment (josesimoes) [inline Tools.BuildTasks-2019/ProcessResourceFiles.cs:896]: what is the purpose of this string here?

File: Tools.BuildTasks-2019/ProcessResourceFiles.cs [hunk 4]
```diff
@@ -891,65 +891,77 @@ public static ResourceTypeDescription ResourceTypeDescriptionFromResourceType(by
...
-                        bitmapValue.RawFormat.Equals(ImageFormat.Jpeg) ||
-                        bitmapValue.RawFormat.Equals(ImageFormat.Gif) ||
-                        bitmapValue.RawFormat.Equals(ImageFormat.Bmp))
-                    {
-                        entry = new BitmapEntry(name, bitmapValue);
-                    }
-                    else
-                    {
-                        // BMP format not supported 
+                    case "String":
```
Comment (josesimoes) [inline Tools.BuildTasks-2019/ProcessResourceFiles.cs:900]: please split the various "case" with an empty line

File: Tools.BuildTasks-2019/ProcessResourceFiles.cs [hunk 5]
```diff
@@ -1314,91 +1326,46 @@ private void Compress1bpp(NanoResourceFile.CLR_GFX_BitmapDescription bitmapDescr
...
 
-                try
-                {
-                    bitmapData = bitmap.LockBits(rect, ImageLockMode.ReadOnly, formatDst);
+                    Rectangle rect = new Rectangle(0, 0, clone.Width, clone.Height);
+                    BitmapData bitmapData = clone.LockBits(rect, ImageLockMode.ReadOnly, PixelFormat.Format16bppRgb565);
 
-                    IntPtr p = bitmapData.Scan0;
-                    data = new byte[bitmapData.Stride * bitmap.Height];
+                    byte[] data = new byte[clone.Width * clone.Height * 2]; // 2 bytes per pixel
```
Comment (josesimoes) [inline Tools.BuildTasks-2019/ProcessResourceFiles.cs:1355]: comment before code line please

**Review by josesimoes — CHANGES_REQUESTED** [review]

**Review by josesimoes — APPROVED** [review]
LGTM!

**Pull request merged**